api/pkg/auth/base.go

```diff
@@ -0,0 +1,81 @@
+// Copyright © 2021 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"net/http"
+	"os"
+
+	"github.com/gorilla/mux"
+	"github.com/gorilla/sessions"
+	"github.com/markbates/goth"
+	"github.com/markbates/goth/gothic"
+	"github.com/markbates/goth/providers/github"
+	"github.com/tektoncd/hub/api/pkg/app"
+	"github.com/tektoncd/hub/api/pkg/auth/provider"
+	auth "github.com/tektoncd/hub/api/pkg/auth/service"
+)
+
+// Auth Provider provides routes for authentication
+// and also defines git providers using goth
+func AuthProvider(r *mux.Router, api app.Config) {
+
+	key := ""            // Replace with your SESSION_SECRET or similar
+	maxAge := 86400 * 30 // 30 days
+	isProd := false      // Set to true when serving over https
```

Comment:
should configure this on the basis of env variable `ENVIRONMENT` ?

Reply:
How about when we read providers from env ?

Reply:
`isProd := false` we will be serving over https right?

Reply:
Are we adding an env in config? How we are changing it?